packages/core/types/index.js

```diff
@@ -299,6 +302,42 @@ export interface Asset extends BaseAsset {
   +stats: Stats;
 }
 
+// TODO: figure out naming
+export interface IConfig {
+  searchPath: FilePath;
+  env: Environment;
+  options: ParcelOptions;
+  resolvedPath: ?FilePath;
+  // $FlowFixMe could be anything
+  result: ?any;
+  resultHash: ?string;
+  includedFiles: Set<FilePath>;
+  watchGlob: ?Glob;
+  devDeps: Map<PackageName, ?string>;
+  pkg: ?PackageJSON;
+
+  setResolvedPath(filePath: FilePath): void;
+  // $FlowFixMe could be anything
+  setResult(result: any): void; // TODO: fix
+  setResultHash(resultHash: string): void;
+  addIncludedFile(filePath: FilePath): void;
+  setDevDep(name: PackageName, version?: Semver): void;
```

Comment:
Can we match these with the methods we have on Asset? `addConnectedFile` and `addDependency`?

Reply:
Yeah `addConnectedFile` and `addIncludedFile` should definitely be consistent. I think I was maybe the one who suggested `connected` vs `included` in the first place. I don't remember perfectly, but I think that was when we were still thinking things like configs would be connected to the asset, where "connected" makes more sense because they are not actually included in the resulting asset. Now that config is connected to config request nodes, "included" seemed like better naming in both cases. I don't think it matters too much, I might be overthinking it. Thoughts?

As for `addDevDep` and `addDependency` I was thinking they are slightly different so it would make sense to have different names. With dependencies we end up resolving the path to that dependency whereas with devDeps we just resolve the version. I don't feel too strongly that they should definitely be named different things though. What do you think?

Reply:
I'm fine with `addConnectedFile` or `addIncludedFile` as long as it's consistent. For dependencies, I can see where the difference is - a dependency actually refers to a `Dependency` type. Maybe we could just spell it out fully: `addDevDependency` (like the package.json field).

Reply:
Updated `Asset` to use `addIncludedFile` instead of `addConnectedFile` and changed `setDevDep` to `addDevDependency`